wordpress-mobile/WordPress-Android
# [Jetpack Content Migration Flow] Refactor - Unify user flags provider

# Description

This PR continues from #17398 by refactoring the user flags provider. With the unified entity provider, the entity-specific logic is extracted to a helper class for the provider-side.

### To test:

#### Precondition:

Turn on the various flags. The following patch may be useful for this purpose:

<details>
<summary>Patch to enable all data migration flags</summary>

```patch
diff --git a/WordPress/build.gradle b/WordPress/build.gradle
index 1a23203a63..cfa2e01bc0 100644
--- a/WordPress/build.gradle
+++ b/WordPress/build.gradle
@@ -120,11 +120,11 @@ android {
         buildConfigField "boolean", "BETA_SITE_DESIGNS", "false"
         buildConfigField "boolean", "JETPACK_POWERED", "true"
         buildConfigField "boolean", "JETPACK_POWERED_BOTTOM_SHEET", "false"
-        buildConfigField "boolean", "JETPACK_SHARED_LOGIN", "false"
-        buildConfigField "boolean", "JETPACK_LOCAL_USER_FLAGS", "false"
-        buildConfigField "boolean", "JETPACK_BLOGGING_REMINDERS_SYNC", "false"
-        buildConfigField "boolean", "JETPACK_READER_SAVED_POSTS", "false"
-        buildConfigField "boolean", "JETPACK_PROVIDER_SYNC", "false"
+        buildConfigField "boolean", "JETPACK_SHARED_LOGIN", "true"
+        buildConfigField "boolean", "JETPACK_LOCAL_USER_FLAGS", "true"
+        buildConfigField "boolean", "JETPACK_BLOGGING_REMINDERS_SYNC", "true"
+        buildConfigField "boolean", "JETPACK_READER_SAVED_POSTS", "true"
+        buildConfigField "boolean", "JETPACK_PROVIDER_SYNC", "true"
         buildConfigField "boolean", "JETPACK_MIGRATION_FLOW", "false"
         buildConfigField "boolean", "JETPACK_FEATURE_REMOVAL_PHASE_ONE", "false"
         buildConfigField "boolean", "JETPACK_FEATURE_REMOVAL_PHASE_TWO", "false"
```
</details>

Log into WordPress:

1. Install the WordPress app flavor from this branch
2. Login to the WordPress app and select a site
3. Make note of (or change some) user flags

Log out of Jetpack and delete all data:

* Force stop the app (if running)
* Go to storage and clear app data

#### Testing steps:

1. Run the Jetpack app from this branch
2. Expect to be logged in automatically
4. Expect to see any changed user flags copied to Jetpack



## Regression Notes
1. Potential unintended areas of impact
No. Existing implementations are changed, but are behind a feature flag.

9. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

10. What automated tests I added (or what prevented me from doing so)
These will be refactored in a later PR to keep the change set manageable.

PR submission checklist:
